code/controllers/CMSMain.php

```diff
@@ -88,8 +97,24 @@ public function init() {
 
 		CMSBatchActionHandler::register('publish', 'CMSBatchAction_Publish');
 		CMSBatchActionHandler::register('unpublish', 'CMSBatchAction_Unpublish');
-		CMSBatchActionHandler::register('delete', 'CMSBatchAction_Delete');
-		CMSBatchActionHandler::register('deletefromlive', 'CMSBatchAction_DeleteFromLive');
+		
+
+		// Check legacy actions
+		$legacy = $this->config()->enabled_legacy_actions;
```

Comment:
Oh right, I remember now.

I tested this, but the problem is the config is global; We'd need to tell OTHER LeftAndMain subclasses which also use batch actions (AssetAdmin) not to add these SiteTree specific batches. I tried to add a "for controller class" config but it ended up as a huge mess and I gave up.

The problem is that the config in place has no allowance for instance specific batch actions. It wouldn't be a problem if the config was on the CONTROLLER, but it's on BatchActionHandler, which is shared for all controllers.

Hell right?